File: apollo-http-cache-api/src/main/java/com/apollographql/apollo/api/cache/http/HttpCacheRecordEditor.java
package com.apollographql.apollo.api.cache.http;

import okio.Sink;
import org.jetbrains.annotations.NotNull;

import java.io.IOException;

public interface HttpCacheRecordEditor {
  @NotNull Sink headerSink();

  @NotNull Sink bodySink();

  void abort() throws IOException;

  void commit() throws IOException;
}

File: apollo-http-cache-api/src/main/kotlin/com/apollographql/apollo/api/cache/http/HttpCacheRecordEditor.kt
package com.apollographql.apollo.api.cache.http

import okio.Sink
import java.io.IOException

interface HttpCacheRecordEditor {
  fun headerSink(): Sink
  fun bodySink(): Sink

  @Throws(IOException::class)
  fun abort()

  @Throws(IOException::class)
  fun commit()
}
